docs/ref/contrib/humanize.txt

```diff
@@ -143,3 +143,5 @@ Examples:
 * ``3`` becomes ``3rd``.
 
 You can pass in either an integer or a string representation of an integer.
+Only non-negative integers are converted. Anything else, including negative
+integers and non-integers, is returned unchanged.
```

Comment:
```suggestion
```
I think instead of this, let's take inspiration from the [`apnumber` docs](https://docs.djangoproject.com/en/5.0/ref/contrib/humanize/#apnumber)

```
Converts a non-negative integer to its ordinal as a string. Otherwise, returns the number.

Examples:

* ``0`` becomes ``0th``.
* ``1`` becomes ``1st``.
* ``2`` becomes ``2nd``.
* ``3`` becomes ``3rd``.
* ``1.5`` becomes ``1.5``.
* ``-4`` becomes ``-4``.
```

Reply:
That's actually trickier than you might think.
`1.5` becomes `"1st"`
`"1.5"` indeed stays as `"1.5"`

Reply:
Sounds like we should also make an update and test cases here 👍

Reply:
I may be overthinking this, but my current suggestion for documentation text start is:
"You can pass in a non-negative number, or a string representation for a non-negative integer. Anything else is returned unchanged".

I don't know how many examples and test cases we want to add. For example, it will work for `Decimal` and `Fraction` instances as well. Then again, we probably don't want to get silly.

Reply:
> You can pass in a non-negative number, or a string representation for a non-negative integer. Anything else is returned unchanged

This doesn't hold true for 1.5 -> "1st"

Reply:
Not sure I follow the logic for ordinals and floating points, but that sounds like another issue. Marking this conversation as resolved.

Reply:
You're adding docs here says that non-integers will be returned unchanged, this is not true. Either this should be removed or you need go handle floats.

Reply:
True, I'll get on removing it

Reply:
Replaced with:
```diff
You can pass in either an integer or a string representation of an integer.
+ Negative integers are returned unchanged.
```